internal/mode/static/policies/observability/generator.go

```diff
@@ -0,0 +1,77 @@
+package observability
+
+import (
+	"fmt"
+	"strings"
+	"text/template"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/helpers"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/policies"
+)
+
+var tmpl = template.Must(template.New("observability policy").Parse(observabilityTemplate))
+
+const observabilityTemplate = `
+{{- if .Tracing }}
+otel_trace {{ .Strategy }};
+  {{- if .Tracing.Context }}
+otel_trace_context {{ .Tracing.Context }};
+  {{- end }}
+  {{- if .Tracing.SpanName }}
+otel_span_name "{{ .Tracing.SpanName }}";
+  {{- end }}
+  {{- range $attr := .Tracing.SpanAttributes }}
+otel_span_attr "{{ $attr.Key }}" "{{ $attr.Value }}";
+  {{- end }}
+  {{- range $attr := .GlobalSpanAttributes }}
+otel_span_attr "{{ $attr.Key }}" "{{ $attr.Value }}";
+  {{- end }}
+{{- end }}
+`
+
+// Generate generates configuration as []byte for an ObservabilityPolicy.
+func Generate(policy policies.Policy, globalSettings *ngfAPI.NginxProxy) []byte {
+	obs := helpers.MustCastObject[*ngfAPI.ObservabilityPolicy](policy)
+
+	var strategy string
+	if obs.Spec.Tracing != nil {
+		switch obs.Spec.Tracing.Strategy {
+		case ngfAPI.TraceStrategyParent:
+			strategy = "$otel_parent_sampled"
+		case ngfAPI.TraceStrategyRatio:
+			strategy = "on"
+			if obs.Spec.Tracing.Ratio != nil {
+				if *obs.Spec.Tracing.Ratio > 0 {
+					strategy = CreateRatioVarName(obs)
+				} else {
+					strategy = "off"
+				}
+			}
+		default:
+			strategy = "off"
+		}
+	}
+
+	var spanAttributes []ngfAPI.SpanAttribute
+	if globalSettings != nil && globalSettings.Spec.Telemetry != nil {
+		spanAttributes = globalSettings.Spec.Telemetry.SpanAttributes
+	}
+
+	fields := map[string]interface{}{
+		"Tracing":              obs.Spec.Tracing,
+		"Strategy":             strategy,
+		"GlobalSpanAttributes": spanAttributes,
+	}
+
+	return helpers.MustExecuteTemplate(tmpl, fields)
+}
+
+// CreateRatioVarName builds a variable name for an ObservabilityPolicy to be used with
+// ratio-based trace sampling.
+func CreateRatioVarName(policy *ngfAPI.ObservabilityPolicy) string {
+	namespace := strings.ReplaceAll(policy.GetNamespace(), "-", "_")
+	name := strings.ReplaceAll(policy.GetName(), "-", "_")
+
+	return fmt.Sprintf("$ratio_%s_%s", namespace, name)
```

Comment:
Could we still get collisons?
```
	name/name-name => $rate_ns_name_name_name_name
	name-name/name => $rate_ns_name_name_name_name
```


could you possibly double check that NGINX doesn't fail to reload when the namespace and name has max allowed length by Kubernetes? (sorry, missed this during prev review)

Reply:
lol you're good at these. Any suggestions? It needs to be a fairly predictable naming system because we build this name in two different places, so indexing probably isn't an option.

Reply:
There don't seem to be any length issues.

Reply:
This is also quite the edge case if someone has resources named like that...

Reply:
> Any suggestions? It needs to be a fairly predictable naming system because we build this name in two different places, so indexing probably isn't an option

what if we use ratio value in the id?

For example, if ratio is 30, we will have $obs_ratio_30 
If multiple policies use the same ratio, it's gonna be ok, since the corresponding split clients can be reused among them.

Reply:
Yep, I was about to comment the same thing

Reply:
> There don't seem to be any length issues.

forgot to mention. if variables become long, NGINX can print warning during reloads 
This was addressed here though https://github.com/nginxinc/nginx-gateway-fabric/blob/f908851578fc963401678d60b3462543e2b60958/internal/mode/static/nginx/config/variable_names.go#L15